app/models/user.rb

```diff
@@ -123,6 +123,16 @@ def censoring?(tag)
     self.censored_tags.exists?(name: tag)
   end
 
+  def censoring_tags?(tags)
+    tags.map! do |tag|
+      if tag.instance_of?(Tag)
+        tag = tag.name
+      end
+    end
```

Comment:
```suggestion
    tag_names = tags.map do |tag|
      tag.instance_of?(Tag) ? tag.name : tag
    end
```
Rubyのmapブロックは最後に評価された値が代入されます。そのため、元のコードだと`tag.instance_of?(Tag)`がfalseを返したときに最後に評価される値がfalseになるので、nilが配列に入って想定してない動きになると思います（別のコードで手元で実験したらそうでした。上の例で違ってたらごめんなさい）。

他のcensoring?等にも言えることなので申し訳ないのですが、ダックタイピングを考えると
``` suggestion
    tag_names = tags.map do |tag|
      tag&.name || tag
    end
```
みたいな設計でもいいです。

Reply:
```suggestion
    tag_names = tags.map do |tag|
      tag.respond_to?(:name) ? tag.name : tag
    end
```

（上のsuggestionはこっちのほうがさらにいいです。他のcensoring?とかも守れてなくて申し訳ない……）